### PR TITLE
test: Add Test Cases for Payment Reconciliation and Related Doctypes

### PR DESCRIPTION
**Pull request by rahulsharma8848**
### Title:
Add Test Cases for Payment Reconciliation and Related Doctypes

### Test Summary:
This test suite adds automated unit tests for the Payment Reconciliation and related doctypes (Payment Reconciliation Allocation, Invoice, Payment, and Record) in ERPNext. It ensures proper save and database behavior, validates static method overrides, and confirms error handling and dimension filter logic. The suite also verifies that override methods and cancel restrictions function as intended.

### Scenarios Covered
Document Operations
test_custom_save_TC_ACC_205
Validates that a Payment Reconciliation document can be created and saved without errors.

test_db_insert_override_TC_ACC_206
Ensures that the custom db_insert method executes without exceptions.

test_db_update_override_TC_ACC_207
Validates that the custom db_update method runs successfully.

Static Method Overrides
test_static_get_list_TC_ACC_208
Verifies the get_list static method returns None as expected.

test_static_get_count_TC_ACC_209
Checks that the get_count method returns None (default stub behavior).

test_static_get_stats_TC_ACC_210
Ensures get_stats returns None, confirming placeholder implementation.

Functional Logic
test_get_queries_for_dimension_filters_TC_ACC_211
Mocks the dimensions fetched and validates filter construction logic when company-linked dimensions are present.

Child Table Doctype Overrides
test_get_list_is_callable_for_allocation_TC_ACC_212
Confirms PaymentReconciliationAllocation.get_list() is callable without exceptions.

test_get_list_is_callable_for_invoice_TC_ACC_213
Validates that PaymentReconciliationInvoice.get_list() executes as expected.

test_get_list_is_callable_for_payment_TC_ACC_214
Ensures PaymentReconciliationPayment.get_list() can be invoked without side effects.

Edge Case Handling
test_on_cancel_raises_error_TC_ACC_215
Tests that cancelling a submitted Payment Reconciliation Record raises a ValidationError, adhering to business rules preventing record cancellation.

### Technology:
Framework: unittest using FrappeTestCase base class

Mocks: unittest.mock.patch for dimension fetching logic

Database: Temporary test records created via Frappe ORM (frappe.get_doc and insert)

### Linked Feature/Bug:
None

### CODE REVIEW

File: erpnext/accounts/doctype/payment_reconciliation_record/test_payment_reconciliation_record.py [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+	def test_on_cancel_raises_error_TC_ACC_215(self):
+		# Create a test PaymentReconciliationRecord
+		doc = frappe.get_doc({"doctype": "Payment Reconciliation Record", "title": "Test Cancellation Block"})
+		doc.insert()
+		doc.submit()
+		doc.reload()
+
+		# Attempt to cancel and assert that an exception is thrown
+		with self.assertRaises(frappe.ValidationError, msg="Cancelling records is not allowed"):
+			doc.cancel()
```
Comment (aasif-patel) [inline erpnext/accounts/doctype/payment_reconciliation_record/test_payment_reconciliation_record.py]: Use assertIn with assertRaises to check exact validation msg as assertRaises is passing the test case in all validations

File: erpnext/accounts/doctype/payment_reconciliation_payment/test_payment_reconciliation_payment.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from erpnext.accounts.doctype.payment_reconciliation_payment.payment_reconciliation_payment import (
+	PaymentReconciliationPayment,
+)
+
+
+class TestPaymentReconciliationPayment(FrappeTestCase):
+	def test_get_list_is_callable_for_payment_TC_ACC_214(self):
+		# Call the static method with dummy args,Since method does nothing, No use of  assert
+		PaymentReconciliationPayment.get_list(args={})
```
Comment (aasif-patel) [inline erpnext/accounts/doctype/payment_reconciliation_payment/test_payment_reconciliation_payment.py]: if it's return nothing then this test case make no sense.

File: erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py [hunk 3]
```diff
@@ -2980,6 +2981,63 @@ def test_payment_reconciliation_record_on_unreconcile_TC_ACC_217(self):
...
+		doc = frappe.get_doc({"doctype": "Payment Reconciliation", "title": "Test Save"})
+		doc.save()  # no assert required
+
+	def test_db_insert_override_TC_ACC_206(self):
+		doc = frappe.get_doc({"doctype": "Payment Reconciliation", "title": "Test Insert"})
+		doc.db_insert()  # no assert required
+
+	def test_db_update_override_TC_ACC_207(self):
+		doc = frappe.get_doc({"doctype": "Payment Reconciliation", "title": "Test Update"})
+		doc.db_update()  # no assert required
```
Comment (aasif-patel) [inline erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py]: ignore the cod cov for this functions this test cases makes no sense as the methods are passing doing nothing

**Review by aasif-patel — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/payment_reconciliation_record/test_payment_reconciliation_record.py (hunk 1 above)
Comment (rahulsharma8848) [reply erpnext/accounts/doctype/payment_reconciliation_record/test_payment_reconciliation_record.py]: updated

File: erpnext/accounts/doctype/payment_reconciliation_payment/test_payment_reconciliation_payment.py (hunk 2 above)
Comment (rahulsharma8848) [reply erpnext/accounts/doctype/payment_reconciliation_payment/test_payment_reconciliation_payment.py]: updated

File: erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py (hunk 3 above)
Comment (rahulsharma8848) [reply erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py]: updated

**Comment by swathibollineni2** [comment]
@rahulsharma8848 below test cases are not found in postgres instance. Kindly  check

TC_ACC_205
TC_ACC_206
TC_ACC_207
TC_ACC_214
